### PR TITLE
Implement LAZ read support based on the laz-perf library

### PR DESCRIPTION
**Pull request by hobu**

### CODE REVIEW

File: laspy/base.py [hunk 1]
```diff
@@ -159,6 +160,42 @@ def point_map(self):
         self.pointfmt = np.dtype([("point", [(str(x.name), x.np_fmt) for x in self.manager.point_format.specs])])
 
         if self.manager.header.version not in ("1.3", "1.4"):
+            if self.compressed:
+                import lazperf
```
Comment (kbevers) [inline laspy/base.py]: Shouldn't you check the status of `HAVE_LAZPERF` before moving on here?

Comment (hobu) [reply laspy/base.py]: doh

File: laspy/__init__.py [hunk 2]
```diff
@@ -1,6 +1,6 @@
 from __future__ import absolute_import
 
-__version__ = '1.6.0'
+__version__ = '1.7.0'
```
Comment (hobu) [inline laspy/__init__.py]: Note I plan to roll this functionality out with a new release. Also included will be something related to #102

File: laspy/base.py [hunk 3]
```diff
@@ -124,11 +129,7 @@ def __init__(self, filename, manager):
 
     def open(self, mode):
         '''Open the file, catch simple problems.'''
-        if (not self.compressed) or self.mode == "r-":
-            try:
-                self.fileref = open(self.filename, mode)
-            except(Exception):
-                raise laspy.util.LaspyException("Error opening file. Do you have the right permissions?")
+        self.fileref = open(self.filename, mode)
```
Comment (hobu) [inline laspy/base.py]: laspy has a very bad habit of catching exceptions for you and swallowing the useful information that might be in them. This block adds no value and takes away from the user's ability to figure out why their file won't open.

File: laspy/base.py [hunk 4]
```diff
@@ -178,14 +215,12 @@ def point_map(self):
...
-                    self._mmap = False
-                    self._pmap = False
-                except(Exception):
-                    raise laspy.util.LaspyException("Error closing mmap")
+            try:
+                self._mmap.flush()
+                self._mmap.close()
+            except(Exception):
+                self._mmap = False
+                self._pmap = False
```
Comment (hobu) [inline laspy/base.py]: same on the exception catching as above. If we can't flush()/close(), there's nothing we can do about it except zero them out, which is what the method does anyway. Telling the user the map wasn't closed() doesn't help them do anything about it.

**Comment by hobu** [comment]
The `lazperf` library works on windows, osx, and linux. Should we consider making it a dependency of laspy?

**Comment by kbevers** [comment]
> The lazperf library works on windows, osx, and linux. Should we consider making it a dependency of laspy?

Is is easy to "pip install"? If not, is it at least less of a hassle to install compared to the current `laszip` approach? I suspect that is is :)

**Comment by hobu** [comment]
It does `pip install` just fine. The issue with it is that it doesn't support LAZ 1.4 data.

**Pull request merged**